src/org/rascalmpl/library/util/ErrorRecovery.rsc

```diff
@@ -17,17 +17,24 @@ module util::ErrorRecovery
 import ParseTree;
 import String;
 
+@synopsis{Library functions for handling parse trees with error nodes.}
+@description{
+When parse functions are called with `errorRecovery=true`, the resulting parse trees can contain error and skipped nodes. For an in-depth description of these nodes see the `ParseTree` module.
+Functions in this module can be used to detect, isolate, and analyze these error nodes.
+Note that these are fairly low-level functions. Better high-level support for error trees will be added to Rascal and its standard library in the future.
+As such, this code should be considered experimental and used with care.
+}
+
 @synopsis{Check if a parse tree contains any error nodes, the result of error recovery.}
 bool hasErrors(Tree tree) = /appl(error(_, _, _), _) := tree;
 
 @javaClass{org.rascalmpl.library.util.ErrorRecovery}
-@synopsis{Find all error productions in a parse tree. The list is created by an outermost visit of the parse tree so if an error tree contains other errors the outermost tree is returned first.}
+@synopsis{Find all error productions in a parse tree. The list is created by an outermost visit of the parse tree so if an error tree contains other errors the outermost tree is returned first.
+Often error trees are highly ambiguous and can contain a lot of error trees. This function is primarily used to analyze small examples as calling this function
```

Comment:
Error trees containing error trees? Is that what this is intended to say, or is it a typo?

Reply:
No this is not a typo. Children of error trees can contain errors themselves. I will explain this in the comment.